baseapp/abci.go

```diff
@@ -312,6 +314,37 @@ func (app *BaseApp) deliverTxWithoutEventHistory(req abci.RequestDeliverTx) (res
 	}
 }
 
+// checkHalt checkes if height or time exceeds halt-height or halt-time respectively.
+func (app *BaseApp) checkHalt(height int64, time time.Time) {
+	var halt bool
+	switch {
+	case app.haltHeight > 0 && uint64(height) > app.haltHeight:
+		halt = true
+
+	case app.haltTime > 0 && time.Unix() > int64(app.haltTime):
```

Comment:
Isn't this invalid? [`time.Unix`](https://pkg.go.dev/time#Unix) requires POSIX epoch timestamp input in the form of `sec` and `nsec` arguments, and returns a `time.Time` which is incomparable with `int64`.
```suggestion
	case app.haltTime > 0 && time.Now().Unix() > int64(app.haltTime):
```
(but I'm uncomfortable with using local time rather than block time for this)

Reply:
Ah.  You were misled by the fact the call wasn't using a static function of the `time` package, because that package was shadowed by `checkHalt`'s `time` argument.  So, it was actually using the argument's instance method, `(t) Unix` (no valid link available, but take a look just below [`(t) UTC`](https://pkg.go.dev/time#Time.UTC).  To avoid this confusion, I've renamed the argument to `tm` instead.